crates/task-impls/src/consensus/mod.rs

```diff
@@ -829,6 +860,46 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>> ConsensusTaskState<TYPES, I>
                     };
                 }
             }
+            HotShotEvent::QuorumVoteSend(vote) => {
+                let Some(proposal) = self.current_proposal.clone() else {
+                    return;
+                };
+                let new_view = proposal.get_view_number() + 1;
+                // In future we can use the mempool model where we fetch the proposal if we don't have it, instead of having to wait for it here
+                // This is for the case where we form a QC but have not yet seen the previous proposal ourselves
+                let should_propose = self.quorum_membership.get_leader(new_view) == self.public_key
+                    && self.consensus.read().await.high_qc.view_number
+                        == proposal.get_view_number();
+
+                if should_propose {
+                    debug!(
+                        "Attempting to publish proposal after voting; now in view: {}",
+                        *new_view
+                    );
+                    let _ = self.publish_proposal(new_view, event_stream.clone()).await;
+                }
+                if proposal.get_view_number() <= vote.get_view_number() {
+                    self.current_proposal = None;
+                }
+            }
+            HotShotEvent::QuorumProposalSend(proposal, _) => {
+                if self
+                    .payload_commitment_and_metadata
+                    .as_ref()
+                    .is_some_and(|p| p.block_view <= proposal.data.get_view_number())
+                {
+                    self.payload_commitment_and_metadata = None;
+                }
```

Comment:
Is it correct in my understanding that we only want to wipe the payload commitment and metadata when we successfully propose? Would this not basically just become stale if we didn't propose again after that?

Reply:
Yes we want to wipe it when we propose.  It is stale for future views, but it is possible another propose task gets spawned after the first proposal (or is already running).  Resetting this could prevent double proposing.  The new variable in the locked state `lastest_proposed_view` is what's really in charge of that, but doing this shouldn't hurt.